compiler/z/codegen/OMRRegisterDependencyStruct.hpp

```diff
@@ -42,59 +42,16 @@ namespace OMR { typedef OMR::Z::RegisterDependencyExt RegisterDependency; }
 #include "infra/Array.hpp"            // for TR_Array
 #include "infra/Assert.hpp"           // for TR_ASSERT
 
-namespace TR { struct RegisterDependency; }
-
 namespace OMR
 {
 namespace Z
 {
-struct RegisterDependencyExt
+struct RegisterDependencyExt : OMR::RegisterDependencyExt
    {
-   uint32_t                   _realRegister:7;
-   uint32_t                   _flags:2;
-   uint32_t                   _virtualRegIndex:23;
```

Comment:
@ymanton @0dvictor @andrewcraik @0xdaryl note that the Z codegen was actually different than all the other codegens in this respect. We used a bitflag to represent a register dependency to save on space. Each register dependency is 32-bits in memory. We did this with a tradeoff to access the real register will require an extra indirection via `cg->getRegisterArray()`

I've changed it now to do it in the same way as Power and x86, though I'm not sure which is the better way as the size of each dependency is now 128-bits on 64-bit platforms (one pointer and two integer fields which will be aligned). So it is a trade off between space vs. time in accessing the real register. Which do we think is better overall?